File: apps/desktop/frontend/src/types/RootSearchSnapshot.ts
import type { NavigationSnapshot, SearchPhase, SearchResult } from "./index";

export interface RootSearchSnapshot
{
    navigation: NavigationSnapshot;
    sessionId: number;
    requestId: number;
    revision: number;
    query: string;
    results: SearchResult[];
    resultRevision: number;
    resultOffset: number;
    totalResults: number;
    phase: SearchPhase;
    error: string | null;
    warnings: string[];
}

File: apps/desktop/shell/src/RootSearchSnapshot.rs
use serde::Serialize;

use crate::{SearchPhase, SearchResult};

#[derive(Debug, Clone, Serialize)]
#[serde(rename_all = "camelCase")]
pub(crate) struct RootSearchSnapshot {
    pub(crate) navigation: crate::NavigationSnapshot,
    pub(crate) session_id: u64,
    pub(crate) request_id: u64,
    pub(crate) revision: u64,
    pub(crate) query: String,
    #[serde(skip_serializing_if = "Option::is_none")]
    pub(crate) results: Option<Vec<SearchResult>>,
    pub(crate) phase: SearchPhase,
    pub(crate) error: Option<String>,
    pub(crate) warnings: Vec<String>,
}
